flow/shared/authenticator.go

```diff
@@ -0,0 +1,218 @@
+package shared
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type AuthenticationConfig struct {
+	DevKeySetJSON string `json:"key_set_json" yaml:"key_set_json" mapstructure:"key_set_json"`
+	Auth0Domain   string `json:"auth0_domain" yaml:"auth0_domain" mapstructure:"auth0_domain"`
+	Auth0ClientID string `json:"auth0_client_id" yaml:"auth0_client_id" mapstructure:"auth0_client_id"`
+	Enabled       bool   `json:"enabled" yaml:"enabled" mapstructure:"enabled"`
+}
+
+type identityProvider struct {
+	keySet      jwk.Set
+	validateOpt jwt.ValidateOption
+	issuer      string
+}
+
+func AuthGrpcMiddleware() ([]grpc.ServerOption, error) {
+	devKeySetJSON, _ := os.LookupEnv("PEERDB_OAUTH_KEY_SET_JSON")
+	auth0Domain, _ := os.LookupEnv("PEERDB_OAUTH_AUTH0DOMAIN")
```

Comment:
nit: don't like the digit in the env var
also could use `GetEnvString`

Reply:
yeah, TODO move to peerdbenv once stabilized